### PR TITLE
Request for c sample, adding updated Engineering sample

### PR DESCRIPTION
**Pull request by m0fff**

### CODE REVIEW

File: PI System/Community Samples/c/omfDemo/omfDemo/sendCert.c [hunk 1]
```diff
@@ -0,0 +1,248 @@
+/*
+
+©2009-2017 OSIsoft, LLC. All Rights Reserved.
```
Comment (dshvidchenko) [inline PI System/Community Samples/c/omfDemo/omfDemo/sendCert.c]: years, copyrights

File: PI System/Community Samples/c/omfDemo/omfDemo/sendToServerC.c [hunk 2]
```diff
@@ -0,0 +1,263 @@
+/*
+
+©2009-2017 OSIsoft, LLC. All Rights Reserved.
```
Comment (dshvidchenko) [inline PI System/Community Samples/c/omfDemo/omfDemo/sendToServerC.c]: years, copyrights

File: PI System/Community Samples/c/omfDemo/omfDemo/sendToServerC.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/*
+
+©2009-2017 OSIsoft, LLC. All Rights Reserved.
```
Comment (dshvidchenko) [inline PI System/Community Samples/c/omfDemo/omfDemo/sendToServerC.h]: years, copyrights

File: PI System/Community Samples/c/omfDemo/omfDemo/sendToServerC.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+for damages of any kind arising from Licensee's use of the OSIsoft Samples OR OTHERWISE, including
+but not limited to direct, indirect, special, incidental and consequential damages, and Licensee
+expressly assumes the risk of all such damages.  FURTHER, THE OSIsoft SAMPLES ARE NOT ELIGIBLE FOR
+SUPPORT UNDER EITHER OSISOFT'S STANDARD OR ENTERPRISE LEVEL SUPPORT AGREEMENTS
+
+*/
+
+/* sendToServerC.h*/
+
+#pragma once
```
Comment (dshvidchenko) [inline PI System/Community Samples/c/omfDemo/omfDemo/sendToServerC.h]: use #ifndef #define

File: PI System/Community Samples/c/omfDemo/omfDemo/testTxMessages.c [hunk 5]
```diff
@@ -0,0 +1,288 @@
+/*
+
+©2009-2017 OSIsoft, LLC. All Rights Reserved.
```
Comment (dshvidchenko) [inline PI System/Community Samples/c/omfDemo/omfDemo/testTxMessages.c]: years, copyrights

File: PI System/Community Samples/c/omfDemo/omfDemo/testTxMessages.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
+/*
+
+©2009-2017 OSIsoft, LLC. All Rights Reserved.
```
Comment (dshvidchenko) [inline PI System/Community Samples/c/omfDemo/omfDemo/testTxMessages.h]: years, copyrights

File: PI System/Community Samples/c/omfDemo/omfDemo/testTxMessages.h [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+PURPOSE and NONINFRINGEMENT. In no event will OSIsoft be liable to Licensee or to any third party
+for damages of any kind arising from Licensee's use of the OSIsoft Samples OR OTHERWISE, including
+but not limited to direct, indirect, special, incidental and consequential damages, and Licensee
+expressly assumes the risk of all such damages.  FURTHER, THE OSIsoft SAMPLES ARE NOT ELIGIBLE FOR
+SUPPORT UNDER EITHER OSISOFT'S STANDARD OR ENTERPRISE LEVEL SUPPORT AGREEMENTS
+
+*/
+
+#pragma once
```
Comment (dshvidchenko) [inline PI System/Community Samples/c/omfDemo/omfDemo/testTxMessages.h]: use #ifndef #define

File: PI System/Community Samples/c/omfDemo/omfDemo/getValuesAndStreams.c [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+		szRet = defineLink_Sources_Targets(&(szVehicles[0][0]), TYPE_VehicleType, NULL, &(szTransmissions[0][0]), TYPE_TransmissionType, NULL, NUM_VEHICLES, STRING_LENGTH_MED);
+		break;
+	}
+
+	return szRet;
+}
+
+char* getValues_EngineCoolantTempData()
+{
+#define NUM_COOLANT_VALUES  4
```
[the remark this answers is not in the document]
Comment (m0fff) [reply PI System/Community Samples/c/omfDemo/omfDemo/getValuesAndStreams.c]: For the sample, including them with the function they are used in aids readability

File: PI System/Community Samples/c/omfDemo/omfDemo/bufferData.h [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+for damages of any kind arising from Licensee's use of the OSIsoft Samples OR OTHERWISE, including
+but not limited to direct, indirect, special, incidental and consequential damages, and Licensee
+expressly assumes the risk of all such damages.  FURTHER, THE OSIsoft SAMPLES ARE NOT ELIGIBLE FOR
+SUPPORT UNDER EITHER OSISOFT'S STANDARD OR ENTERPRISE LEVEL SUPPORT AGREEMENTS
+
+*/
+
+/* bufferData.h */
+
+#pragma once
```
[the remark this answers is not in the document]
Comment (m0fff) [reply PI System/Community Samples/c/omfDemo/omfDemo/bufferData.h]: Unnecessary change, adds no value, #pragma while not standard is supported by a range of compilers.

File: PI System/Community Samples/c/omfDemo/omfDemo/bufferData.h [hunk 10]
```diff
@@ -0,0 +1,38 @@
+/*
+
+©2009-2017 OSIsoft, LLC. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (m0fff) [reply PI System/Community Samples/c/omfDemo/omfDemo/bufferData.h]: updated

**Comment by m0fff** [comment]
copyright updated.

**Pull request closed without merging**